### DISCUSSION TITLE
only supports Suspense but doesn't support server components?

### DISCUSSION THREAD

**Discussion by hyjoong** [TanStack/query#5729] [General] [answered]
I recently saw that the `react-query-nextjs` package was supported for release a `Suspense Streaming`.
You have to declare `'use client'` because you have to fill the client with QueryCache, and eventually you don't support server components, do you think it's applicable even if the page is designed as data fetching on RSC?

**Comment by TkDodo** [comment, marked as answer]
If you are using react query, you are likely using it because it's an async state manager that keeps your data up-to-date over time. To do that, it requires a filled cache on the client. And yes, it requires client components to do that, because it runs fetches from the browser to achieve that.

If you want a pure server solution, where your JS only runs on the server, you don't need react-query, because it wouldn't bring anything beneficial to the table.

https://tkdodo.eu/blog/you-might-not-need-react-query
